dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
+        final Group group = groupService.findByName(context, Group.ANONYMOUS);
+        context.restoreAuthSystemState();
+        String token = getAuthToken(admin.getEmail(), password);
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isOk());
+
+        getClient(authToken).perform(
+                delete("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isUnprocessableEntity());
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void deleteGroupForbiddenTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
+
+            context.commit();
+
+            parentGroup = context.reloadEntity(parentGroup);
+
+            String adminToken = getAuthToken(admin.getEmail(), password);
+            String authToken = getAuthToken(eperson.getEmail(), password);
+
+            getClient(adminToken).perform(
+                    get("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isOk());
+
+            getClient(authToken).perform(
+                    delete("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isForbidden());
+
+            getClient(adminToken).perform(
+                    get("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isOk());
+
+        } finally {
+            if (parentGroup != null) {
+                GroupBuilder.deleteGroup(parentGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void deleteGroupNotFoundTest() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        context.commit();
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(authToken).perform(
+                delete("/api/eperson/groups/" + UUID.randomUUID())
+        ).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void getGroupObjectCommunityTest() throws Exception {
+        CommunityService communityService
+                = ContentServiceFactory.getInstance().getCommunityService();
+        Community community = null;
+        Group adminGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+            community = communityService.create(null, context);
+            adminGroup = communityService.createAdministrators(context, community);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
```

Comment:
Same as above. should use GroupBuilder, missing a context.restoreAuthSystemState(); and same question about commit() and reloadEntity() as above.